### PR TITLE
Enable Kura API baseline analysis

### PR DESCRIPTION
**Pull request by cdealti**
**Note:** The bnd-baseline-maven-plugin does not work with Maven 3.0.x (currently used in our Hudson builds).
**Note 2:** The Hudson build currently fails switching to Maven 3.3.9 (see https://bugs.eclipse.org/bugs/show_bug.cgi?id=514168) while it succeeds locally or on Travis CI.
**Update**: The [Maven 3.3.9 Hudson](https://hudson.eclipse.org/kura/job/kura-develop-maven-3.3.9/19/) build now succeeds. See bug resolution details at the above link.

* Annotated org.eclipse.kura.api with @ProviderType and @ConsumerType
  OSGi annotations
* Tagged org.eclipse.kura.api with @noimplement and @noextend Javadoc tags
* Tagged new API packages with @since Javadoc tags
* Fixed versions in existing @since Javadoc tags.
  Now the version matches the API Bundle-Version.
* Enabled CI API analysis through bnd-baseline-maven-plugin
* Updated MANIFESTs as suggested by bnd-baseline-maven-plugin
* TODO: add bnd-baseline-maven-plugin 3.4.0-SNAPSHOT to build/test CQ
* TODO: upgrade bnd-baseline-maven-plugin to 3.4.0 when released

This reverts the revert merge #897 of #685

Signed-off-by: Cristiano De Alti <cristiano.dealti@eurotech.com>

### CODE REVIEW

**Comment by cdealti** [comment]
Also see the [Semantic Versioning Wiki page](https://github.com/eclipse/kura/wiki/Kura-Semantic-Versioning)

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/bluetooth/listener/AdvertisingReportRecord.java [hunk 1]
```diff
@@ -11,6 +11,12 @@
  *******************************************************************************/
 package org.eclipse.kura.bluetooth.listener;
 
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * @noextend
```
Comment (amitjoy) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/bluetooth/listener/AdvertisingReportRecord.java]: @noextend This class is not intended to be subclassed by clients.

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/bluetooth/listener/BluetoothAdvertisementData.java [hunk 2]
```diff
@@ -14,6 +14,12 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * @noextend
```
Comment (amitjoy) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/bluetooth/listener/BluetoothAdvertisementData.java]: @noextend This class is not intended to be subclassed by clients.

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/firewall/FirewallAutoNatConfig.java [hunk 3]
```diff
@@ -13,13 +13,16 @@
 package org.eclipse.kura.net.firewall;
 
 import org.eclipse.kura.net.NetConfig;
+import org.osgi.annotation.versioning.ProviderType;
 
 /**
  * Represents an automatic NAT configuration
  *
  * @author eurotech
```
Comment (amitjoy) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/firewall/FirewallAutoNatConfig.java]: As you have previously deleted @authour tags in this PR, you can also remove this.

**Comment by cdealti** [comment]
@dwoodard1 I believe this could be merged.
If you merge it, the develop build on Hudson will fail once but it can then be fixed:

https://bugs.eclipse.org/bugs/show_bug.cgi?id=514168

**Pull request merged**